eng/tools/spec-gen-sdk-runner/src/utils.ts

```diff
@@ -1,11 +1,36 @@
-import { spawn, spawnSync } from "node:child_process";
+import { spawn, spawnSync, exec } from "node:child_process";
 import path from "node:path";
 import fs from "node:fs";
 import { LogLevel, logMessage } from "./log.js";
+import { promisify } from "node:util";
 
 type Dirent = fs.Dirent;
 
-// Common function to find files recursively with case-insensitive matching
+export const execAsync = promisify(exec);
+
+/**
+ * Reset unstaged changes in a git repository
+ * @param repoPath The path to the git repository
+ * @returns A promise that resolves when the reset is complete
+ */
+export async function resetGitRepo(repoPath: string): Promise<void> {
+  try {
+    const { stderr } = await execAsync("git clean -fd && git reset --hard HEAD", {
```

Comment:
```suggestion
    const { stderr } = await execAsync("git clean -fdx && git reset --hard HEAD", {
```
Add x so ignored files also get deleted. Will reset HEAD be correct? It will only work if we haven't create a new branch and committed those changes, or do we just need a clean git status?

Reply:
'reset' HEAD is needed to remove any modifications to the tracked files.

Reply:
Added 'x'.